### PR TITLE
Set Mongo host from container host

### PR DESCRIPTION
**Pull request by ghub-real**
#### What type of PR is this?
- Bug Fix

#### What does this PR do / why is it needed ?
Fix failing test - with the mongo hostname also parameterized in addition to port number.

### CODE REVIEW

**Review by rafaelbey — DISMISSED** [review]

**Comment by rafaelbey** [comment]
We are missing replacing localhost here: https://github.com/finos/legend-engine/blob/master/legend-engine-xt-nonrelationalStore-mongodb-executionPlan-test/src/main/java/org/finos/legend/engine/plan/execution/stores/mongodb/test/shared/MongoDBCommands.java#L132

File: legend-engine-xt-nonrelationalStore-mongodb-executionPlan-test/src/main/resources/core_mongodb_execution_test/test_setup/testSetupDSL.txt [hunk 1]
```diff
@@ -89,7 +89,7 @@ MongoDBConnection test::mongodb::testConnection
 {
     database: userDatabase;
     store: test::mongodb::mydatabase;
-    serverURLs: [localhost:__PORT_TOKEN__];
+    serverURLs: [__HOST_TOKEN__:__PORT_TOKEN__];
```
Comment (rafaelbey) [inline legend-engine-xt-nonrelationalStore-mongodb-executionPlan-test/src/main/resources/core_mongodb_execution_test/test_setup/testSetupDSL.txt:92]: All this works from connectivity perspective, but when the hostname have characters like '-', the parser fails.  This is a limitation on what kind of hostname you will be supporting, and one that we should fill the gap to fix some of the e2e testing we do.

Comment (ghub-real) [reply legend-engine-xt-nonrelationalStore-mongodb-executionPlan-test/src/main/resources/core_mongodb_execution_test/test_setup/testSetupDSL.txt:92]: Fixed the parser to accept - as well, but we might have to support a wider set of values potentially.

**Review by rafaelbey — APPROVED** [review]

**Review by rafaelbey — APPROVED** [review]

**Pull request merged**